tests/unit/s2n_cert_validation_callback_test.c

```diff
@@ -444,6 +434,119 @@ int main(int argc, char *argv[])
 
             EXPECT_EQUAL(data.invoked_count, 1);
         }
+
+        /* Async callback is invoked on the client after receiving the server's certificate */
+        for (int i = 0; i < s2n_array_len(test_cases); i++) {
+            DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+            EXPECT_NOT_NULL(config);
+            EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key));
+            EXPECT_SUCCESS(s2n_config_set_verification_ca_location(config, S2N_DEFAULT_TEST_CERT_CHAIN, NULL));
+            EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "default"));
```

Comment:
It's best avoid using the default security policy for testing unless you want to specifically test the behavior of the default security policy, which isn't really the case for these tests. Probably best to pin it:
```suggestion
            EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "20240501"));
```

Reply:
It also might be good to iterate over a TLS 1.3 and a TLS 1.2 security policy for these tests, since the cert_recv logic changes a bit:
https://github.com/aws/s2n-tls/blob/e4a5a74b0492767ac996e93a7257d533087015dd/tls/s2n_server_cert.c#L25

Just want to make sure that the stuffer reset logic works properly in all cases.